### DISCUSSION TITLE
vt52 and /opt/pidp10/bin/vt52

### DISCUSSION THREAD

**Discussion by statespacedev** [PDP-10/-READ--THIS-#52] [Q&A] [answered]
hi all - am in 'beginners brain' mode - could use a quick pointer to 'proper place' to ask questions, or does 'discussions' right here work ok?:)
am in tops-10 using sos - terminal is /opt/pidp10/bin/vt52
is there a way to type 'rubout' and 'linefeed'? :)
hmm, maybe there is a 'better' terminal program from 'keyboard perspective' should use instead of /opt/pidp10/bin/vt52?

update - seems can get something like 'rubout' via ctrl-backspace - at least at monitor prompt, but not so much in sos, which is weird:) and shift-enter might be something at least close to 'linefeed', but not exactly, at least in sos

**Comment by larsbrinkhoff** [comment]
For TOPS-10 question, I think the Usenet group alt.sys.pdp10, or the PiDP-10 group - https://groups.google.com/g/pidp-10 - is your best bet.

**Reply by statespacedev** [reply]
just answered my top question while googling 'vt52 emulator with line feed':)

ctrl-j is line feed! working beautifully in sos:)

hi lars, thanks, understood - have thought of both those too, makes sense. for some reason just felt like jumping in here and trying something new, hope it's ok for everyone?

why? well, usenet works and will always have a place in the heart, going back thirty five years now, but actually seems kind of depressing, or something? and the pidp-10 google group is great, watch it daily - could be more about the 'front panel kit'? dunno - we'll see - early days!:) for me, would be happy spending more time here 'in github' - just seems attractive at the moment? 

am hitting ctrl-j for line feed in those last lines here
![PXL_20240713_142519328](https://github.com/user-attachments/assets/85574a6c-f107-48e8-a814-27f53f9e3be4)

**Comment by statespacedev** [comment]
ok, another mystery solved:)
'delete' key is in fact working for 'rubout/delete' - octal 177
so would say have sos and ddt 'fully in action again' here:) should be good to go
was a bit confusing because it echoes what got rubbed out behind \ delimiter
but it is in fact rubbing out!
one other little discovery - ctrl-l is form feed!
will do a little blog post putting all this in a nutshell and will share here

**Comment by david-moon** [comment]
On Jul 13, 2024, at 11:46 AM, Noah Smith ***@***.***> wrote:
> 
> ctrl-l is form feed!

Same as it ever was.

**Reply by statespacedev** [reply]
hi david, just curious - was it ctrl-l on the real vt52 too? been forty years, don't remember - but was looking at the keyboard layout yesterday and wondering about form feed
![image](https://github.com/user-attachments/assets/6a1a27e8-03ea-4d1f-ab6a-ff5643d6606d)
and here's the 'bell'
https://youtu.be/bAafRXddfxc?si=J6SnM9HYCSHArtPH
btw another random factoid - noticed this morning that ddt circa early eighties uses 'backspace / ctrl-h' key as kinda the inverse of 'linefeed / ctrl-j' - closes current location and moves 'up', rather than 'down'. that's the idea at least - but what am seeing here right now at this very moment is that ctrl-h works fine, but backspace key does not! so ctrl-j and ctrl-h it is:)

**Comment by statespacedev** [comment, marked as answer]
quick summary of learnings - enough to get sos and ddt 'fully into action', completely able to do everything in the gorin book, which was the objective. threw in a nutshell with the sos commands. !minimalism!:)
https://statespace.dev/vt52.html
